crates/svm-storage2/src/layout/builder.rs

```diff
@@ -0,0 +1,45 @@
+use super::DataLayout;
+
+/// Specifies the fixed-sized variables of an application.
+pub struct DataLayoutBuilder {
+    vars: Vec<(u32, u32)>,
+
+    next_offset: u32,
+}
+
+/// `DataLayoutBuilder` represents the fixed-sized variables (storage) of an application.
+impl DataLayoutBuilder {
+    pub fn new() -> Self {
+        Self::with_capacity(0)
+    }
+
+    /// New instance, initialized with the total number of variables.
+    pub fn with_capacity(capacity: usize) -> Self {
+        Self {
+            vars: Vec::with_capacity(capacity),
+            next_offset: 0,
+        }
+    }
+
+    pub fn extend_from_slice(&mut self, slice: &[u32]) {
+        for len in slice.iter() {
+            self.add_var(*len);
+        }
+    }
+
+    /// Adds a the next variable's.
+    ///
+    /// The `var_id` equals to the previous one plus one.
+    /// The starting offset is right after where the previous variable ended.
```

Comment:
We could have managed the builder with having `vars` represented as 
`Vec<u32`> and computing the offsets in the `build` method.

I chose to make it so that `vars` field will hold more info and the `build` 
the implementation will be remain logic:

```rust
pub fn build(self) -> DataLayout {
    DataLayout { vars: self.vars }
}
```

Reply:
Similarly, the `DataLayout` could contain only offsets, but then I'd have to diff two adjacent variables when computing a variable's length. I prefer to add more data and have the final code simplified.